### DISCUSSION TITLE
porting r2ModMan to a new title?

### DISCUSSION THREAD

**Discussion by StephenJTurnbull** [ebkr/r2modmanPlus#534] [Q&A]
Is there any chance of bringing something like this to Thea / Thea2 and the not yet released re-mastering of Masters Of Magic on the Honey Engine by the indie studio MuHa games? 

Currently I am trying to use DnSpy to do compile level edits to the game and DnSpy is tripping over some namespace redundancy/clobbering  and a few definitions that seam to be lost in the de-compiling. 

Some work around that allowed for smaller sharper code changes would be appreciated (or just tip sand tricks for massaging life into a incomplete decompile), if your tool work on that Unity game Engine as well. (and it interests you at all). 

If not, Have fun in the projects you do choose to take on.

**Comment by ebkr** [comment]
Pretty much the only requirements for adding a game are:

1. Is a unity game
2. Mods use BepInEx
3. Has a Thunderstore community

The manager definitely isn’t oriented towards development though

**Reply by StephenJTurnbull** [reply]
Thea 2 has a mod via workshop tool,  it just doesn't allow modders to change the parts of the game I want to change. I looked into using DnSpy to make changes, and I can, but the part of the code I want to change doesn't get de-compiled correctly, so I can't make any changes that way, LiSpy fails for much the same reason.  

but Thea2 doesn't have a Thunderstore community,  it is a Unity game, and BepInEx probably could make modes for it in addition to the mods that already exist via workshop. 
Is there anybody to petition adding a thunderstore community for Thea2?

**Reply by ebkr** [reply]
Yeah a Thunderstore community is something that would have to be made. 

The requirements are generally just that you have some active modders in a discord server willing to upload to TS, although the final decision lies with Mythic. There may be some other requirements I’m unaware of.

What are you expecting r2modman to do in this regard? It doesn’t decompile the game, it simply just installs BepInEx which would have to be uploaded to the TS community for the games should one be created.

**Reply by StephenJTurnbull** [reply]
I am just looking for a way to get into the group movement point logic of Thea 2 and adjust it. DnSpy failed me. I know that r2Mod does allow code to be re-written. A better decompiler would also solve my problems/help me get to my goal. but decompiling is hard, it's going to be hard for DnSpy or anything else.

**Reply by ebkr** [reply]
So r2modman doesn’t really do anything in that regard. BepInEx is what allows you to mod in the way the supported games allow.

r2modman just moves compiled DLLs and any associated assetbundles/pictures/other into the correct folder, however the DLLs are all built using BepInEx and are in no way the actual game assets.

Neither the manager, nor BepInEx, allow you to decompile the game.

Since DnSpy is failing you, an alternative could be dotPeek.

**Reply by StephenJTurnbull** [reply]
Thank you!
